File: backend/app/models/offer.py
"""Daily Offer models (FEATURES_COMMUNITY_PLUS · Daily Offer)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class Offer(BaseModel):
    id: str
    title: str
    brand: str | None = None
    image_url: str | None = None
    discount_label: str | None = None
    affiliate_url: str  # already attribution-tagged by the backend
    topics: list[str] = Field(default_factory=list)

File: app/lib/data/models/offer.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'offer.freezed.dart';
part 'offer.g.dart';

/// A curated/affiliate deal for the Newsroom "Offers" strip
/// (FEATURES_COMMUNITY_PLUS · Daily Offer). [affiliateUrl] is already
/// attribution-tagged by the backend.
@freezed
abstract class Offer with _$Offer {
  const factory Offer({
    required String id,
    required String title,
    String? brand,
    @JsonKey(name: 'image_url') String? imageUrl,
    @JsonKey(name: 'discount_label') String? discountLabel,
    @JsonKey(name: 'affiliate_url') required String affiliateUrl,
    @Default(<String>[]) List<String> topics,
  }) = _Offer;

  factory Offer.fromJson(Map<String, dynamic> json) => _$OfferFromJson(json);
}
